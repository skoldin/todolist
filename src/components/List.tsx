import React from "react";
import ListItem from "./ListItem";

interface ListItems {
  listItems: object[];
  updateItem: (item: Item) => void;
  deleteItem: (id: string) => void;
}

const List: React.FunctionComponent<ListItems> = ({
  listItems,
  updateItem,
  deleteItem
}) => {
  if (!listItems.length) {
    return null;
    ``;
  }

  return (
    <ul className="list-group">
      {/* TODO: typecheck this */}
      {listItems.map((item: any) => {
        const { id } = item;

        return (
          <ListItem
            item={item}
            key={id}
            updateItem={updateItem}
            deleteItem={deleteItem}
          />
        );
      })}
    </ul>
  );
};

export default List;
