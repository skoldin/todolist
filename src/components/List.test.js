import React from "react";
import List from "./List";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

const props = {
  listItems: [
    { id: "1", name: "test 1", done: false },
    { id: "2", name: "test 2", done: true }
  ]
};

let Component = renderer.create(<List {...props} />);
let wrapper;

beforeEach(() => {
  wrapper = shallow(<List {...props} />);
});

it("render matches the snapshot", () => {
  expect(Component.toJSON()).toMatchSnapshot();
});

it("displays 2 items", () => {
  expect(wrapper.find("ListItem")).toHaveLength(2);
});
