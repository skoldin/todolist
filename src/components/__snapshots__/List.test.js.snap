// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render matches the snapshot 1`] = `
<ul
  className="list-group"
>
  <li
    className="list-group-item d-flex justify-content-between align-items-center"
  >
    <div
      className="custom-control custom-checkbox"
    >
      <input
        className="custom-control-input"
        defaultChecked={false}
        id="1"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="custom-control-label"
        htmlFor="1"
        style={Object {}}
      >
        test 1
      </label>
    </div>
    <button
      className="btn btn-link"
      onClick={[Function]}
    >
      Delete
    </button>
  </li>
  <li
    className="list-group-item d-flex justify-content-between align-items-center"
  >
    <div
      className="custom-control custom-checkbox"
    >
      <input
        className="custom-control-input"
        defaultChecked={true}
        id="2"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="custom-control-label"
        htmlFor="2"
        style={
          Object {
            "color": "#999",
            "textDecoration": "line-through",
          }
        }
      >
        test 2
      </label>
    </div>
    <button
      className="btn btn-link"
      onClick={[Function]}
    >
      Delete
    </button>
  </li>
</ul>
`;
